jlm/llvm/opt/ScalarEvolution.cpp

```diff
@@ -15,6 +16,81 @@
 
 namespace jlm::llvm
 {
+
+class ScalarEvolution::Context final
+{
+public:
+  ~Context() = default;
+
+  Context() = default;
+
+  Context(const Context &) = delete;
+
+  Context(Context &&) = delete;
+
+  Context &
+  operator=(const Context &) = delete;
+
+  Context &
+  operator=(Context &&) = delete;
+
+  void
+  InsertChrec(const rvsdg::ThetaNode & thetaNode, std::unique_ptr<SCEVChainRecurrence> & chrec)
+  {
+    ChrecMap_[&thetaNode].insert(std::unique_ptr<SCEVChainRecurrence>(chrec.release()));
```

Comment:
Instead of creating a new unique pointer from a released pointer, you can use std::move